ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/worldview/VerkleWorldState.java

```diff
@@ -366,6 +292,69 @@ private void updateAccountStorageState(
     }
   }
 
+  private void updateState(
+      final Address accountKey,
+      final VerkleTrie stateTrie,
+      final Optional<VerkleWorldStateKeyValueStorage.Updater> maybeStateUpdater,
+      final HasherContext hasherContext,
+      final VerkleWorldStateUpdateAccumulator worldStateUpdater) {
+
+    final VerkleEntryFactory verkleEntryFactory = new VerkleEntryFactory(hasherContext.hasher());
+
+    generateAccountValues(
+        accountKey,
+        verkleEntryFactory,
+        maybeStateUpdater,
+        worldStateUpdater.getAccountsToUpdate().get(accountKey));
+
+    generateCodeValues(
+        accountKey,
+        verkleEntryFactory,
+        maybeStateUpdater,
+        worldStateUpdater.getCodeToUpdate().get(accountKey));
+
+    generateStorageValues(
+        accountKey,
+        verkleEntryFactory,
+        maybeStateUpdater,
+        worldStateUpdater.getStorageToUpdate().get(accountKey));
+
+    verkleEntryFactory
+        .getKeysForRemoval()
+        .forEach(
+            key -> {
+              System.out.println("remove key " + key);
+              stateTrie.remove(key);
+            });
+    verkleEntryFactory
+        .getNonStorageKeyValuesForUpdate()
+        .forEach(
+            (key, value) -> {
+              System.out.println("add key " + key + "  leaf value " + value);
+              stateTrie.put(key, value);
+            });
+    verkleEntryFactory
+        .getStorageKeyValuesForUpdate()
+        .forEach(
+            (storageSlotKey, pair) -> {
+              var storageAccountUpdate = worldStateUpdater.getStorageToUpdate().get(accountKey);
```

Comment:
I don't think this one is still needed as we already checked here https://github.com/hyperledger/besu/pull/7594/files#diff-a8c0068954d44abe33b85a9a6c2b70309adb959255e417350abe54c858574b62R267 but maybe I missing something ?

Reply:
Not sure I understand. So the link you posted is when the key values are generated. Here we commit them to the trie in one go.
On generation we only have the `storageSlotKey` and corresponding key/value. We don't have the `storageUpdate` ref where we should save the prior value

Reply:
I was talking about this check (==null) 
https://github.com/hyperledger/besu/pull/7594/files/2d12b659a7478f155f9a7be52e1dfed6dfcb3c7d#diff-a8c0068954d44abe33b85a9a6c2b70309adb959255e417350abe54c858574b62R341

when we are generating the key we already check if the collection is empty, so imo when we enter this loop we cannot have a null

Reply:
yes you're right that we if there are elements we can't have null. I added it based on a defensive approach since we don't know how the programmer will generate keys and values. What if it's based on another object?
If someone changes the code above, then we will not break here. I would prefer to leave it